src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const listOfNewStates = [];
+
+  const oneAction = (task) => {
+    switch (task.type) {
+      case 'addProperties':
+        Object.assign(newState, task.extraData);
+        break;
+      case 'removeProperties':
+        task.keysToRemove.forEach((key) => delete newState[key]);
+        break;
+      case 'clear':
+        Object.keys(newState).forEach((key) => delete newState[key]);
+    }
+
+    listOfNewStates.push({ ...newState });
+  };
```

Comment:
The 'oneAction' function should not directly mutate a shared 'newState' object. Instead, create a new state copy within the function to ensure that each action results in a distinct state clone.